src/util/ogr_source_util.cpp

```diff
@@ -447,41 +449,125 @@ bool OGRSourceUtil::readAttributesIntoCollection(AttributeArrays &attributeArray
 			time2Index = i;
+		}
 	}
 
 	if(time1Index == -1)
 		throw OperatorException("OGR Source: time1 attribute not found.");
-	
+
 	if(timeSpecification == TimeSpecification::START_DURATION || timeSpecification == TimeSpecification::START_END)
 	{
 		if(time2Index == -1)
-			throw OperatorException("OGR Source: time1 attribute not found.");
+			throw OperatorException("OGR Source: time2 attribute not found.");
 	}
+
+    //try filtering the attributes via OGRLayer, so we don't have to do it manually in readTimeIntoCollection
+	timeAlreadyFiltered = trySettingTemporalFilter(layer, rect);
+}
+
+bool OGRSourceUtil::trySettingTemporalFilter(OGRLayer *layer, const QueryRectangle &rect){
+	// DateTime and Date are the supported field data type!
+	// Both fields have to available and of type DateTime/Date, because only START_END as TimeSpecification is supported.
+	// That's because I couldn't find a way to express in the sql like query (time_attribute + time_duration).
+	if(timeSpecification != TimeSpecification::START_END)
+        return false;
+
+    // can try to set the filter without the fields being of type Date/DateTime. But that is unsafe and
+    // can result in empty collections. So the filter could be set on string data fields and still be a good query.
+    // but both attributes still have to be present and be of a Date/DateTime structure (as a string).
+    bool hasSaveTypes = (time1Type == OGRFieldType::OFTDateTime || time1Type == OGRFieldType::OFTDate) &&
+                           (time2Type == OGRFieldType::OFTDateTime || time2Type == OGRFieldType::OFTDate);
+
+    // if this parameter is true, ignore if we have save types. e.g. WFS would still work, if the data on the wfs server
+    // is Date/DateTime because the filtering happens there, but the file send to us will have String as data type,
+    // because the WFS driver can not send it while keeping Date/DateTime datatypes.
+	const bool forceOGRTimeFiltering = params.get("force_ogr_time_filter", false).asBool();
+
+    if(!forceOGRTimeFiltering && !hasSaveTypes)
+        return false;
+
+    // now check if the fields are strings if force is used. Testing against Integer, Double wouldn't make sense
+    if(!hasSaveTypes && forceOGRTimeFiltering && (time1Type != OGRFieldType::OFTString || time2Type != OGRFieldType::OFTString))
+        return false;
+
+	//create TimeDate representation of rect.t1 and rect.t2
+	std::string rect_time_String_1 = "\'";
+	rect_time_String_1 += rect.toIsoString(rect.t1);
+	rect_time_String_1 += "\'";
+
+	std::string rect_time_string_2 = "\'";
+	rect_time_string_2 += rect.toIsoString(rect.t2);
+	rect_time_string_2 += "\'";
+
+	std::stringstream filter_stream;
+    // Create SQL like comparison to check if attributes validity intersects with the query rectangles times:
+    // (feature.t1 >= rect.t1 || feature.t2 >= rect.t1) && (feature.t1 <= rect.t2 || feature.t2 <= rect.t2)
+    filter_stream << "( " << time1Name << " >= " << rect_time_String_1;
```

Comment:
is it safe to just use the user specified values here or do we need to sanitize the input?

Reply:
At this point we know that `time1Name`/`time2Name` are existing attributes. We also know they have a supported data type (or are Strings if the filtering is forced). So I think this is fine.